cmake/CMakeLists.txt

```diff
@@ -44,18 +63,17 @@ if (CMAKE_USE_PTHREADS_INIT)
   add_definitions(-DHAVE_PTHREAD)
 endif (CMAKE_USE_PTHREADS_INIT)
 
-if (MSVC)
-  if (ZLIB)
-    set(HAVE_ZLIB 1)
-    find_path(ZLIB_INCLUDE_DIRECTORIES zlib.h ${protobuf_SOURCE_DIR})
-    find_library(ZLIB_LIBRARIES zdll ${protobuf_SOURCE_DIR})
-  else (ZLIB)
-    set(HAVE_ZLIB 0)
-  endif (ZLIB)
-else (MSVC)
+if (ZLIB)
   find_package(ZLIB)
```

Comment:
I do not feel any difficulty on the Windows with MSVC.
You can easily find pre-builded and installed version of the ZLIB library with a standard CMake module FindZLIB.cmake.
To find the library, you can use standard hints:
cmake -D[CMAKE_PREFIX_PATH](http://www.cmake.org/cmake/help/v3.3/variable/CMAKE_PREFIX_PATH.html?highlight=cmake_prefix_path)=<path_to_zlib_install>
(cmake -D[CMAKE_INSTALL_PREFIX](http://www.cmake.org/cmake/help/v3.3/variable/CMAKE_INSTALL_PREFIX.html?highlight=cmake_install_prefix)=<path_to_zlib_install>)
Or more specific hint:
  cmake -DZLIB_ROOT=<path_to_zlib_install>
See more in module documentation [FindZLIB](http://www.cmake.org/cmake/help/v3.3/module/FindZLIB.html).